os/include/tinyara/pm/pm.h

```diff
@@ -176,115 +176,35 @@
 #warning "This logic needs to be extended"
 #endif
 
-/* State changes then occur when the weight activity account crosses
- * threshold values for certain periods of time (time slice count).
- *
- * CONFIG_PM_xxxENTER_THRESH is the threshold value for entering state xxx.
- * CONFIG_PM_xxxENTER_COUNT is the count for entering state xxx.
- *
- * Resuming to normal state, on the other hand, is usually immediate and
- * controlled by wakeup conditions established by the platform.  The PM
- * module only recommends reduced power states.
- */
-
-#ifndef CONFIG_PM_IDLEENTER_THRESH
-#define CONFIG_PM_IDLEENTER_THRESH    1	/* <=1: Essentially no activity */
-#endif
-
-#ifndef CONFIG_PM_IDLEEXIT_THRESH
-#define CONFIG_PM_IDLEEXIT_THRESH     2	/* >=2: Active */
-#endif
-
-#if CONFIG_PM_IDLEENTER_THRESH >= CONFIG_PM_IDLEEXIT_THRESH
-#error "Must have CONFIG_PM_IDLEENTER_THRESH < CONFIG_PM_IDLEEXIT_THRESH"
-#endif
-
-#ifndef CONFIG_PM_IDLEENTER_COUNT
-#define CONFIG_PM_IDLEENTER_COUNT     30	/* Thirty IDLE slices to enter
-											 * IDLE mode from normal
-											 */
-#endif
-
-#ifndef CONFIG_PM_STANDBYENTER_THRESH
-#define CONFIG_PM_STANDBYENTER_THRESH 1	/*  <=1: Essentially no activity */
-#endif
-
-#ifndef CONFIG_PM_STANDBYEXIT_THRESH
-#define CONFIG_PM_STANDBYEXIT_THRESH  2	/* >=2: Active */
-#endif
-
-#if CONFIG_PM_STANDBYENTER_THRESH >= CONFIG_PM_STANDBYEXIT_THRESH
-#error "Must have CONFIG_PM_STANDBYENTER_THRESH < CONFIG_PM_STANDBYEXIT_THRESH"
-#endif
-
-#ifndef CONFIG_PM_STANDBYENTER_COUNT
-#define CONFIG_PM_STANDBYENTER_COUNT  50	/* Fifty IDLE slices to enter
-											 * STANDBY mode from IDLE
-											 */
-#endif
-
-#ifndef CONFIG_PM_SLEEPENTER_THRESH
-#define CONFIG_PM_SLEEPENTER_THRESH   1	/*  <=1: Essentially no activity */
-#endif
-
-#ifndef CONFIG_PM_SLEEPEXIT_THRESH
-#define CONFIG_PM_SLEEPEXIT_THRESH    2	/* >=2: Active */
-#endif
-
-#if CONFIG_PM_SLEEPENTER_THRESH >= CONFIG_PM_SLEEPEXIT_THRESH
-#error "Must have CONFIG_PM_SLEEPENTER_THRESH < CONFIG_PM_SLEEPEXIT_THRESH"
-#endif
-
-#ifndef CONFIG_PM_SLEEPENTER_COUNT
-#define CONFIG_PM_SLEEPENTER_COUNT    70	/* 70 IDLE slices to enter SLEEP
-											 * mode from STANDBY
-											 */
-#endif
-
 /* Defines max length of device driver name for PM callback. */
 #define MAX_PM_CALLBACK_NAME    32
 
 /****************************************************************************
  * Public Types
  ****************************************************************************/
 
-/* This enumeration provides all power management states.  Receipt of the
- * state indication is the state transition event.
+/* Defines the normal state for PM framework */
+#define PM_NORMAL		0
+
+/* State changes when occur when the weight accumulator count crosses
+ * threshold values for certain periods of time (time slice count).
+ *
+ * CONFIG_PM_STATE_SWITCH_THRESH is the count threshold value for entering next sleep state xxx.
```

Comment:
The naming for these macros is bit confusing. 
It is difficult to understand the difference between STATE_SWITCH_THRESH and STATE_ENTER_THRESH.
Ideally, STATE_ENTER_THRESH and STATE_EXIT_THRESH are based on activity count, whereas 
STATE_SWITCH_THRESH is based on time slice. So, it is better to change STATE_SWITCH_THRESH to indicate the time slice aspect. Something like STATE_SWITCH_DURATION would be better.